### PR TITLE
linuxPackages.evdi: unstable-2021-06-11 -> unstable-2021-07-07

### LINKED ISSUE #132318: NixOS 21.05: evdi fails to build with latest nixpkgs update (displaylink dependency)
### Describe the bug

A related chunk of log of `nixos-rebuild`:

```
/build/source/module/evdi_modeset.c:331:19: error: initialization of 'void (*)(struct drm_plane *, struct drm_atomic_state *)' from incompatible pointer type 'void (*)(struct drm_plane *, struct drm_plane_state *)' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
  331 |  .atomic_update = evdi_plane_atomic_update,
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~
/build/source/module/evdi_modeset.c:331:19: note: (near initialization for 'evdi_plane_helper_funcs.atomic_update')
/build/source/module/evdi_modeset.c:332:16: error: 'drm_gem_fb_prepare_fb' undeclared here (not in a function); did you mean 'drm_gem_fb_create'?
  332 |  .prepare_fb = drm_gem_fb_prepare_fb
      |                ^~~~~~~~~~~~~~~~~~~~~
      |                drm_gem_fb_create
/build/source/module/evdi_modeset.c:336:19: error: initialization of 'void (*)(struct drm_plane *, struct drm_atomic_state *)' from incompatible pointer type 'void (*)(struct drm_plane *, struct drm_plane_state *)' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
  336 |  .atomic_update = evdi_cursor_atomic_update,
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/module/evdi_modeset.c:336:19: note: (near initialization for 'evdi_cursor_helper_funcs.atomic_update')
cc1: all warnings being treated as errors
make[4]: *** [/nix/store/zbc1cq2y4dkgfhxjmv7q5441v0jviyrb-linux-5.13.6-dev/lib/modules/5.13.6/source/scripts/Makefile.build:273: /build/source/module/evdi_modeset.o] Error 1
make[3]: *** [/nix/store/zbc1cq2y4dkgfhxjmv7q5441v0jviyrb-linux-5.13.6-dev/lib/modules/5.13.6/source/Makefile:1853: /build/source/module] Error 2
make[2]: *** [/nix/store/zbc1cq2y4dkgfhxjmv7q5441v0jviyrb-linux-5.13.6-dev/lib/modules/5.13.6/source/Makefile:220: __sub-make] Error 2
make[2]: Leaving directory '/nix/store/zbc1cq2y4dkgfhxjmv7q5441v0jviyrb-linux-5.13.6-dev/lib/modules/5.13.6/build'
make[1]: *** [Makefile:77: module] Error 2
make[1]: Leaving directory '/build/source/module'
make: *** [Makefile:8: all] Error 2
```

### Steps To Reproduce
1. Update `nixpkgs` channel to https://releases.nixos.org/nixos/21.05/nixos-21.05.1970.11c662074e2 of 31 July 2021 (previous working state is https://releases.nixos.org/nixos/21.05/nixos-21.05.1523.b2f87e0043a of 17 July 2021)
2. Run `sudo nixos-rebuild boot`

### Expected behavior
NixOS rebuilds successfully, DisplayLink works.

### Notify maintainers

- `linuxPackages.evdi` @eyJhb 
- `displaylink` @nshalman @abbradar @peterhoeg @eyJhb

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.12.15, NixOS, 21.05.1523.b2f87e0043a (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.12`
 - channels(root): `"nixos, nixos-unstable"`
 - channels(wenzel): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- linuxPackages.evdi
- displaylink
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by Artturin**
###### Motivation for this change
Closes #132318
Just the patches aren't enough to fix the build now even on nixos-unstable

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
Result of `nixpkgs-review pr 132599` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.evdi</li>
    <li>linuxPackages_4_14.evdi</li>
    <li>linuxPackages_4_4.evdi</li>
    <li>linuxPackages_4_9.evdi</li>
    <li>linuxPackages_latest-libre.evdi</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>displaylink</li>
    <li>linuxPackages.evdi</li>
    <li>linuxPackages_4_19.evdi</li>
    <li>linuxPackages_5_12.evdi</li>
    <li>linuxPackages_5_13.evdi</li>
    <li>linuxPackages_5_4.evdi</li>
    <li>linuxPackages_hardened.evdi</li>
    <li>linuxPackages_latest_hardened.evdi</li>
    <li>linuxPackages_lqx.evdi</li>
    <li>linuxPackages_testing_bcachefs.evdi</li>
    <li>linuxPackages_xanmod.evdi</li>
    <li>linuxPackages_zen.evdi</li>
  </ul>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 132599` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.evdi</li>
    <li>linuxPackages_4_14.evdi</li>
    <li>linuxPackages_4_4.evdi</li>
    <li>linuxPackages_4_9.evdi</li>
    <li>linuxPackages_hardkernel_4_14.evdi</li>
    <li>linuxPackages_hardkernel_latest.evdi</li>
    <li>linuxPackages_latest-libre.evdi</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>displaylink</li>
  </ul>
</details>
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>linuxPackages.evdi (linuxPackages_5_10.evdi)</li>
    <li>linuxPackages_4_19.evdi</li>
    <li>linuxPackages_5_12.evdi</li>
    <li>linuxPackages_5_13.evdi (linuxPackages_latest.evdi)</li>
    <li>linuxPackages_5_4.evdi</li>
    <li>linuxPackages_hardened.evdi</li>
    <li>linuxPackages_latest_hardened.evdi</li>
    <li>linuxPackages_lqx.evdi</li>
    <li>linuxPackages_testing_bcachefs.evdi</li>
    <li>linuxPackages_xanmod.evdi</li>
    <li>linuxPackages_zen.evdi</li>
  </ul>
</details>

**Comment by Artturin** [comment]
> Result of `nixpkgs-review pr 132599` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 7 packages marked as broken and skipped:
> 1 package failed to build:
> 
>     * displaylink
> 
> 
> 11 packages built:
> 
>     * linuxPackages.evdi (linuxPackages_5_10.evdi)
> 
>     * linuxPackages_4_19.evdi
> 
>     * linuxPackages_5_12.evdi
> 
>     * linuxPackages_5_13.evdi (linuxPackages_latest.evdi)
> 
>     * linuxPackages_5_4.evdi
> 
>     * linuxPackages_hardened.evdi
> 
>     * linuxPackages_latest_hardened.evdi
> 
>     * linuxPackages_lqx.evdi
> 
>     * linuxPackages_testing_bcachefs.evdi
> 
>     * linuxPackages_xanmod.evdi
> 
>     * linuxPackages_zen.evdi

Could you post the log?

**Comment by siraben** [comment]
Ah I didn't download the needed zip file

File: pkgs/os-specific/linux/evdi/default.nix [hunk 1]
```diff
@@ -2,28 +2,15 @@
 
 stdenv.mkDerivation rec {
   pname = "evdi";
-  version = "unstable-2021-06-11";
+  version = "unstable-2021-07-7";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/evdi/default.nix:5]:
```suggestion
  version = "unstable-2021-07-07";
```

File: pkgs/os-specific/linux/evdi/default.nix [hunk 2]
```diff
@@ -2,28 +2,15 @@
...
-    rev = "65e12fca334f2f42396f4e8d16592d53cab34dd6";
-    sha256 = "sha256-81IfdYKadKT7vRdkmxzfGo4KHa4UJ8uJ0K6djQCr22U=";
+    rev = "b0b2c80eb63f9b858b71afa772135f434aea192a";
+    sha256 = "sha256-io+CbZovGjEJjwtmARFH23Djt933ONoHMDoea+i6xFo=";
   };
 
-  # Linux 5.13 support
-  # The patches break compilation for older kernels
-  patches = lib.optional (kernel.kernelAtLeast "5.13") [
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/evdi/default.nix:17]: Please remove the unused fetchpatch from inputs.

**Comment by unclechu** [comment]
Should I expect this to be backported to 21.05?

**Comment by SuperSandro2000** [comment]
If someone does it manually 😂

**Comment by unclechu** [comment]
``` sh
nix-shell -p nix-review --run 'NIXPKGS_ALLOW_UNFREE=1 nixpkgs-review pr 132599'
```

> 5 packages marked as broken and skipped:
linuxPackages-libre.evdi linuxPackages_4_14.evdi linuxPackages_4_4.evdi linuxPackages_4_9.evdi linuxPackages_latest-libre.evdi

> 12 packages built:
displaylink linuxPackages.evdi linuxPackages_4_19.evdi linuxPackages_5_12.evdi linuxPackages_5_13.evdi linuxPackages_5_4.evdi linuxPackages_hardened.evdi linuxPackages_latest_hardened.evdi linuxPackages_lqx.evdi linuxPackages_testing_bcachefs.evdi linuxPackages_xanmod.evdi linuxPackages_zen.evdi

**Comment by Artturin** [comment]
#132668

**Comment by unclechu** [comment]
> If someone does it manually

I will make an MR then.

**Comment by unclechu** [comment]
Oh, nevermind, I’m too late :slightly_smiling_face:

**Pull request merged**